src/features/devices/models/device-additions.ts

```diff
@@ -59,8 +59,13 @@ export const addToModel = (
 	// so it's still provisioning.
 	const isPreProvisioning: AndNode = [
 		'And',
-		isOverallOffline,
+		['Equals', ['ReferencedField', 'device', 'is online'], ['Boolean', false]],
 		['NotExists', ['ReferencedField', 'device', 'last connectivity event']],
+		[
+			'Equals',
+			['ReferencedField', 'device', 'api heartbeat state'],
+			['EmbeddedText', 'unknown'],
+		],
```

Comment:
This change really just does whatever the comment above says...
The reasoning being that, in order for any device to have any heartbeat status != unknown, it means that the supervisor has managed to reach the API at least once, which means that the device is not really any more in pre-provisioning state.

Initial PR that introduces the previous logic
See: https://github.com/balena-io/balena-api/pull/2657/files